### PR TITLE
Add invalidBeforeL and invalidHereAfterL functions

### PR DESCRIPTION
**Pull request by newhoggy**
# Description

Lenses for `ValidityInterval`.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [ ] Self-reviewed the diff

### CODE REVIEW

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs [hunk 1]
```diff
@@ -118,3 +120,23 @@ evalBalanceTxBody ::
 evalBalanceTxBody pp lookupKeyRefund lookupDRepRefund isRegPoolId utxo txBody =
   getConsumedValue pp lookupKeyRefund lookupDRepRefund utxo txBody
     <-> getProducedValue pp isRegPoolId txBody
+
+-- | Lens to access the 'invalidBefore' field of a 'ValidityInterval' as a 'StrictMaybe SlotNo'.
+invalidBeforeL :: Lens' L.ValidityInterval (StrictMaybe SlotNo)
```
Comment (lehins) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs:125]: Copy paste error
```suggestion
invalidBeforeL :: Lens' ValidityInterval (StrictMaybe SlotNo)
```

Comment (newhoggy) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs]: Fixed.  Thanks!

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs [hunk 2]
```diff
@@ -32,6 +32,8 @@ module Cardano.Ledger.Api.Tx.Body (
   AllegraEraTxBody,
   vldtTxBodyL,
   ValidityInterval (..),
+  invalidBeforeL,
+  invalidHereAfterL,
```
Comment (lehins) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs:36]: Could you please add these two lenses to the changelog?

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs [hunk 3]
```diff
@@ -118,3 +122,23 @@ evalBalanceTxBody ::
 evalBalanceTxBody pp lookupKeyRefund lookupDRepRefund isRegPoolId utxo txBody =
   getConsumedValue pp lookupKeyRefund lookupDRepRefund utxo txBody
     <-> getProducedValue pp isRegPoolId txBody
+
+-- | Lens to access the 'invalidBefore' field of a 'ValidityInterval' as a 'StrictMaybe SlotNo'.
+invalidBeforeL :: Lens' ValidityInterval (StrictMaybe SlotNo)
```
Comment (lehins) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs]: We can switch to using `Maybe`
```suggestion
invalidBeforeL :: Lens' ValidityInterval (Maybe SlotNo)
```

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs [hunk 4]
```diff
@@ -118,3 +122,23 @@ evalBalanceTxBody ::
...
+invalidBeforeL = lens g s
+  where
+    g :: ValidityInterval -> StrictMaybe SlotNo
+    g (ValidityInterval a _) = a
+
+    s :: ValidityInterval -> StrictMaybe SlotNo -> ValidityInterval
+    s (ValidityInterval _ b) a = ValidityInterval a b
+
+-- | Lens to access the 'invalidHereAfter' field of a 'ValidityInterval' as a 'StrictMaybe SlotNo'.
+invalidHereAfterL :: Lens' ValidityInterval (StrictMaybe SlotNo)
```
Comment (lehins) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx/Body.hs]: Ditto
```suggestion
invalidHereAfterL :: Lens' ValidityInterval (Maybe SlotNo)
```

**Review by lehins — APPROVED** [review]
Looks good. Thanks

File: libs/cardano-ledger-api/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,9 @@
 # Version history for `cardano-ledger-api`
 
+## 1.7.1.0
+
+* Add `invalidBeforeL` and `invalidHereAfterL`
+
 ## 1.7.0.0
```
Comment (lehins) [inline libs/cardano-ledger-api/CHANGELOG.md]: We add new version as part of the post-release process.
```suggestion
## 1.7.0.0

* Add `invalidBeforeL` and `invalidHereAfterL`
```

**Pull request merged**